### PR TITLE
[Cards] New toggle-button example

### PR DESCRIPTION
**Pull request by galiak11**
Adding a code sample and a screenshot for a Toggle Button example on material.io.

**Code Sample:**
```
  override func collectionView(
    _ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath
  ) -> UICollectionViewCell {

    let cell = collectionView.dequeueReusableCell(
      withReuseIdentifier: String(describing: ToggleButtonCell.self), for: indexPath)
    guard let cardCell = cell as? ToggleButtonCell else { return cell }

    let imagedata = dataSource[indexPath.item]
    cardCell.applyTheme(withScheme: containerScheme) // apply the Material theme to the cards
    cardCell.setCardImage(named: imagedata.image)    // set the card's image based on the datasource
    cardCell.setCustomToggleButton()                 // set the custom selected & unselected icons

    cardCell.isSelectable = true                     // enable toggle button behavior in the card
    cardCell.isSelected = imagedata.selected         // set the default state of the card
    if imagedata.selected {
      collectionView.selectItem(at: indexPath, animated: true, scrollPosition: [])
    }

    cardCell.isAccessibilityElement = true           // make sure the card is accessible
    cardCell.accessibilityLabel = imagedata.accessibilityLabel
    return cardCell
  }
```
**Screenshot:**

CardsToggleButtonExample

### CODE REVIEW

**Comment by romoore** [comment]
Are those heart icons at the correct scale? The screenshot makes them look a little blurry.

File: components/Cards/examples/resources/CardsAssets.xcassets/selected.imageset/Contents.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "selected.png",
+      "scale" : "1x"
+    },
+    {
+      "idiom" : "universal",
+      "scale" : "2x"
```
Comment (romoore) [inline components/Cards/examples/resources/CardsAssets.xcassets/selected.imageset/Contents.json]: Looks like the 2x and 3x assets are missing?

**Comment by yarneo** [comment]
I believe the requestor has already assets for images for the cards. Could we possibly ping them to make sure we are providing the right assets?

File: components/Cards/examples/resources/CardsAssets.xcassets/selected.imageset/Contents.json (hunk 1 above)
Comment (galiak11) [reply components/Cards/examples/resources/CardsAssets.xcassets/selected.imageset/Contents.json]: Do you think these are necessary? We should also consider the increase in the library size.

Comment (romoore) [reply components/Cards/examples/resources/CardsAssets.xcassets/selected.imageset/Contents.json]: If there is no way to use `MDCIcons`, then these assets are necessary for the examples. Examples resources do not increase the size of the libraries for clients, but do increase the repository size. I think it's a balance between required images to produce the example versus reusing existing images. 

In any case, please either use SVGs that Xcode can use to generate PNGs or provide PNGs for all scales. So far I believe the project's norm is to include PNGs, so those should be preferred over moving to SVG.

Comment (galiak11) [reply components/Cards/examples/resources/CardsAssets.xcassets/selected.imageset/Contents.json]: Done

File: components/Cards/examples/supplemental/ToggleButtonCell.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      .withRenderingMode(.alwaysTemplate)
+    let unselectedIcon = UIImage(named: "unselected", in: bundle, compatibleWith: nil)?
+      .withRenderingMode(.alwaysTemplate)
+
+    setImage(selectedIcon, for: .selected)
+    setImage(unselectedIcon, for: .normal)
+    setImageTintColor(.white, for: .normal)
+
+    isSelectable = true
+  }
```
Comment (jverkoey) [inline components/Cards/examples/supplemental/ToggleButtonCell.swift:61]: I'd expect this code and all other initialization to be done in the initializer of the cell; is there a technical reason why we should do so as separate methods?

**Review by jverkoey — APPROVED** [review]
LGTM, but please consider moving the common initialization logic into the cell's initializer; we don't want to encourage non-performant patterns of cell allocation and reuse.

**Comment by jverkoey** [comment]
Please also update the screenshot of the PR to reflect the higher quality assets.

File: components/Cards/examples/supplemental/ToggleButtonCell.swift (hunk 2 above)
Comment (galiak11) [reply components/Cards/examples/supplemental/ToggleButtonCell.swift:61]: Done.

File: components/Cards/examples/resources/CardsAssets.xcassets/selected.imageset/Contents.json (hunk 1 above)
Comment (galiak11) [reply components/Cards/examples/resources/CardsAssets.xcassets/selected.imageset/Contents.json]: Toggle icons are now using @1, @2 & @3 sizes.

**Pull request merged**